### PR TITLE
Made the cost info easier to read

### LINKED ISSUE #2336: [Feature Request]: Make the cost info easier to read
### Is your feature request related to a problem? Please describe.

The `chat_result.cost` is a tuple which users can't tell immediately what the two elements mean. For example,
```python
({'gpt-3.5-turbo-0125': {'completion_tokens': 328,
                         'cost': 0.0015555,
                         'prompt_tokens': 2127,
                         'total_tokens': 2455},
  'total_cost': 0.0015555},
 {'gpt-3.5-turbo-0125': {'completion_tokens': 328,
                         'cost': 0.0015555,
                         'prompt_tokens': 2127,
                         'total_tokens': 2455},
  'total_cost': 0.0015555})
```

### Describe the solution you'd like

It's better to make it easier to read. For example, using a dict.

### Additional context

`gather_usage_summary` in `autogen.agentchat.utils`

### PR DESCRIPTION
**Pull request by Hk669**
## Why are these changes needed?

- Updated the function `gathe_usage_summary` to return a dictionary containing two dictionaries: `total_usage_summary` and `actual_usage_summary`, instead of returning a tuple.

## Related issue number
Closes #2336 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by Hk669** [comment]
@microsoft-github-policy-service agree

File: autogen/agentchat/utils.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def aggregate_summary(usage_summary: Dict[str, Any], agent_summary: Dict[str, An
             aggregate_summary(total_usage_summary, agent.client.total_usage_summary)
             aggregate_summary(actual_usage_summary, agent.client.actual_usage_summary)
 
-    return total_usage_summary, actual_usage_summary
+    return {"total_usage_summary": total_usage_summary, "actual_usage_summary": actual_usage_summary}
```
Comment (sonichi) [inline autogen/agentchat/utils.py:80]: Would it be more clear if we say "usage_including_cached_inference" and "usage_excluding_cached_inference" ?
cc @kevin666aa @qingyun-wu

File: autogen/agentchat/utils.py [hunk 2]
```diff
@@ -26,14 +26,14 @@ def consolidate_chat_info(chat_info, uniform_sender=None) -> None:
...
-def gather_usage_summary(agents: List[Agent]) -> Tuple[Dict[str, any], Dict[str, any]]:
+def gather_usage_summary(agents: List[Agent]) -> Dict[Dict[str, any], Dict[str, any]]:
     r"""Gather usage summary from all agents.
 
     Args:
         agents: (list): List of agents.
 
     Returns:
-        tuple: (total_usage_summary, actual_usage_summary)
+        dictionary: (total_usage_summary, actual_usage_summary)
```
Comment (sonichi) [inline autogen/agentchat/utils.py:36]: Improve the docstr here by adding an example and necessary explanation.

**Review by sonichi — COMMENTED** [review]
Thanks. I have some suggestions.

File: autogen/agentchat/utils.py (hunk 1 above)
Comment (kevin666aa) [reply autogen/agentchat/utils.py:80]: Yes, I think that is better. How about the names in class ChatHistory, should we use `ChatHistory.cost`?

**Review by kevin666aa — CHANGES_REQUESTED** [review]
Please check all places that uses `gather_usage_summary`, there should be a notebook that also uses this function?

**Comment by Hk669** [comment]
@kevin666aa got stuck at a point, if you don't mind, could you please ping me up at discord. username: `hrushikesh_d`

**Comment by kevin666aa** [comment]
> username

Sure!

**Comment by sonichi** [comment]
How's this PR going? I'll need to address this problem asap. I may make a PR myself if there is no update with this PR by tomorrow.

**Comment by Hk669** [comment]
> i need to update all the way from oai/client.py because, it has this method print_usage_summary
if you can give me some specific details about the modification, would be greatly appreciated. thanks @sonichi

would appreciate if you can help me with some more information. @sonichi @kevin666aa

File: autogen/agentchat/chat.py [hunk 3]
```diff
@@ -25,9 +25,9 @@ class ChatResult:
     """The chat history."""
     summary: str = None
     """A summary obtained from the chat."""
-    cost: tuple = None  # (dict, dict) - (total_cost, actual_cost_with_cache)
-    """The cost of the chat. a tuple of (total_cost, total_actual_cost), where total_cost is a
-       dictionary of cost information, and total_actual_cost is a dictionary of information on
+    cost: Dict[dict, dict] = None  # {dict, dict} - {usage_including_cached_inference, usage_excluding_cached_inference}
```
Comment (sonichi) [inline autogen/agentchat/chat.py:28]:
```suggestion
    cost: Dict[str, dict] = None  # keys: "usage_including_cached_inference", "usage_excluding_cached_inference"
```

File: autogen/agentchat/chat.py [hunk 4]
```diff
@@ -25,9 +25,9 @@ class ChatResult:
...
+    """The cost of the chat. a dictionary of (usage_including_cached_inference, usage_excluding_cached_inference), where usage_including_cached_inference is a
```
Comment (sonichi) [inline autogen/agentchat/chat.py:29]: Pleaser revise accordingly

File: autogen/agentchat/utils.py [hunk 5]
```diff
@@ -26,33 +26,42 @@ def consolidate_chat_info(chat_info, uniform_sender=None) -> None:
             ), "llm client must be set in either the recipient or sender when summary_method is reflection_with_llm."
 
 
-def gather_usage_summary(agents: List[Agent]) -> Tuple[Dict[str, any], Dict[str, any]]:
+def gather_usage_summary(agents: List[Agent]) -> Dict[Dict[str, any], Dict[str, any]]:
```
Comment (sonichi) [inline autogen/agentchat/utils.py:29]:
```suggestion
def gather_usage_summary(agents: List[Agent]) -> Dict[Dict[str, Dict], Dict[str, Dict]]:
```

File: autogen/agentchat/utils.py [hunk 6]
```diff
@@ -26,33 +26,42 @@ def consolidate_chat_info(chat_info, uniform_sender=None) -> None:
...
     r"""Gather usage summary from all agents.
 
     Args:
         agents: (list): List of agents.
 
     Returns:
-        tuple: (total_usage_summary, actual_usage_summary)
+        dictionary: {usage_including_cached_inference, usage_excluding_cached_inference}
```
Comment (sonichi) [inline autogen/agentchat/utils.py:36]: Please revise accordingly.

File: autogen/agentchat/utils.py [hunk 7]
```diff
@@ -26,33 +26,42 @@ def consolidate_chat_info(chat_info, uniform_sender=None) -> None:
...
     ```python
-    total_usage_summary = {
-        "total_cost": 0.0006090000000000001,
-        "gpt-35-turbo": {
-                "cost": 0.0006090000000000001,
-                "prompt_tokens": 242,
-                "completion_tokens": 123,
-                "total_tokens": 365
+    {
+        usage_including_cached_inference : {
```
Comment (sonichi) [inline autogen/agentchat/utils.py:42]:
```suggestion
        "usage_including_cached_inference" : {
```

File: autogen/agentchat/utils.py [hunk 8]
```diff
@@ -26,33 +26,42 @@ def consolidate_chat_info(chat_info, uniform_sender=None) -> None:
...
+            "total_cost": 0.0006090000000000001,
+            "gpt-35-turbo": {
+                    "cost": 0.0006090000000000001,
+                    "prompt_tokens": 242,
+                    "completion_tokens": 123,
+                    "total_tokens": 365
+            },
+
+        usage_excluding_cached_inference : {
```
Comment (sonichi) [inline autogen/agentchat/utils.py:51]: Please revise accordingly.

File: autogen/agentchat/utils.py [hunk 9]
```diff
@@ -26,33 +26,42 @@ def consolidate_chat_info(chat_info, uniform_sender=None) -> None:
...
+            "total_cost": 0.0006090000000000001,
+            "gpt-35-turbo": {
+                    "cost": 0.0006090000000000001,
+                    "prompt_tokens": 242,
+                    "completion_tokens": 123,
+                    "total_tokens": 365
         }
```
Comment (sonichi) [inline autogen/agentchat/utils.py:58]: Missing a }

File: notebook/agentchat_cost_token_tracking.ipynb [hunk 10]
```diff
@@ -494,8 +494,9 @@
     }
    ],
    "source": [
-    "total_usage_summary, actual_usage_summary = gather_usage_summary([assistant, ai_user_proxy, user_proxy])\n",
-    "total_usage_summary"
+    "usage_summary = gather_usage_summary([assistant, ai_user_proxy, user_proxy])\n",
+    "usage_summary[\"usage_including_cached_inference\"]\n",
+    "usage_summary[\"usage_excluding_cached_inference\"]"
```
Comment (sonichi) [inline notebook/agentchat_cost_token_tracking.ipynb:499]: Remove this line to match the execution result.

File: autogen/agentchat/chat.py [hunk 11]
```diff
@@ -25,10 +25,12 @@ class ChatResult:
...
     summary: str = None
     """A summary obtained from the chat."""
-    cost: tuple = None  # (dict, dict) - (total_cost, actual_cost_with_cache)
-    """The cost of the chat. a tuple of (total_cost, total_actual_cost), where total_cost is a
-       dictionary of cost information, and total_actual_cost is a dictionary of information on
-       the actual incurred cost with cache."""
+    cost: Dict[str, dict] = None  # keys: "usage_including_cached_inference", "usage_excluding_cached_inference"
+    """The cost of the chat.
+       The value for each usage type is a dictionary containing cost information for that specific type.
+           - "usage_including_cached_inference": Cost information on the actual incurred cost with cache.
```
Comment (sonichi) [inline autogen/agentchat/chat.py:31]:
```suggestion
           - "usage_including_cached_inference": Cost information on the total usage, including the tokens in cached inference.
```

File: autogen/agentchat/chat.py [hunk 12]
```diff
@@ -25,10 +25,12 @@ class ChatResult:
...
+           - "usage_excluding_cached_inference": Cost information on the incurred cost without cache.
```
Comment (sonichi) [inline autogen/agentchat/chat.py:32]:
```suggestion
           - "usage_excluding_cached_inference": Cost information on the usage of tokens, excluding the tokens in cache. No larger than "usage_including_cached_inference".
```

**Review by Hk669 — COMMENTED** [review]
updated the doc string as proposed.

**Comment by sonichi** [comment]
@kevin666aa could you review again? The PR is blocked by the change request.

**Review by kevin666aa — APPROVED** [review]
LGTM!

**Comment by giorgossideris** [comment]
Looks good to me too!

**Review by giorgossideris — APPROVED** [review]

**Pull request merged**